ports/esp32/modmachine.c

```diff
@@ -146,30 +147,36 @@ static void machine_sleep_helper(wake_type_t wake_type, size_t n_args, const mp_
         esp_sleep_enable_timer_wakeup(((uint64_t)expiry) * 1000);
     }
 
-    #if !CONFIG_IDF_TARGET_ESP32C3
-
+    #if SOC_PM_SUPPORT_EXT0_WAKEUP
     if (machine_rtc_config.ext0_pin != -1 && (machine_rtc_config.ext0_wake_types & wake_type)) {
-        esp_sleep_enable_ext0_wakeup(machine_rtc_config.ext0_pin, machine_rtc_config.ext0_level ? 1 : 0);
+        esp_sleep_enable_ext0_wakeup(
+            machine_rtc_config.ext0_pin, 
+            machine_rtc_config.ext0_level ? ESP_EXT0_WAKEUP_ANY_HIGH : ESP_EXT0_WAKEUP_ALL_LOW);
```

Comment:
The ext0 wakeup is configured differently to ext1, this argument is only pin and level (high/low). See https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/sleep_modes.html#_CPPv428esp_sleep_enable_ext0_wakeup10gpio_num_ti

(So this line can stay the same). This is why the CI build is failing.